firestore-bigquery-export/scripts/import/src/run.ts

```diff
@@ -0,0 +1,167 @@
+import { cpus } from "os";
+import * as firebase from "firebase-admin";
+import { pool } from "workerpool";
+
+import { CliConfig } from "./types";
+import { parseConfig } from "./config";
+
+import {
+  ChangeType,
+  FirestoreBigQueryEventHistoryTracker,
+  FirestoreDocumentChangeEvent,
+} from "@firebaseextensions/firestore-bigquery-change-tracker";
+
+/**
+ * Import data from a collection group in parrallel using workers.
+ */
+async function processCollectionGroup(config: CliConfig): Promise<number> {
+  const maxWorkers = Math.ceil(cpus().length / 2);
+  const workerPool = pool(__dirname + "/worker.js", {
+    maxWorkers,
+    forkOpts: {
+      env: {
+        PROJECT_ID: config.projectId,
+        GOOGLE_CLOUD_PROJECT: config.projectId,
+        GCLOUD_PROJECT: config.projectId,
+        ...process.env,
+      },
+    },
+  });
+
+  const query = firebase
+    .firestore()
+    .collectionGroup(config.sourceCollectionPath);
+
+  const partitionsList = query.getPartitions(config.batchSize);
+
+  let total = 0;
+  let partitions = 0;
+
+  while (true) {
+    const inProgressTasks =
+      workerPool.stats().activeTasks + workerPool.stats().pendingTasks;
+    if (inProgressTasks >= maxWorkers) {
+      await new Promise((resolve) => setTimeout(resolve, 150));
+      continue;
+    }
+
+    // @ts-ignore, iterator not typed correctly.
+    const { value: partition, done } = await partitionsList.next();
+    if (done || !partition) {
+      break;
+    }
+
+    partitions++;
+
+    const query = partition.toQuery();
+
+    const serializedQuery = {
+      startAt: query._queryOptions.startAt,
+      endAt: query._queryOptions.endAt,
+      limit: query._queryOptions.limit,
+      offset: query._queryOptions.offset,
+    };
+
+    workerPool
+      .exec("processDocuments", [serializedQuery, config])
+      .then((count) => {
+        total += count;
+        console.log(`${total} documents processed`);
+      })
+      .catch((error) => {
+        // TODO handle, what should happen on error?
```

Comment:
Maybe identify the specific documents that weren't processed (from `serializedQuery`). The script could auto-retry, or output a list of queries that failed (so that someone could maybe manually re-run the script for those?)

Reply:
I like this idea of potentially retrying, for now I have added a message to describe the issue and for a list of queried documents that are needed for manual insertion.

Adding a retry could add a bit complexity, would we prefer to have this for the initial version or create a separate issue to handle in the future?